test/test_videoapi.py

```diff
@@ -57,34 +57,66 @@ def test_frame_reading(self):
             full_path = os.path.join(VIDEO_DIR, test_video)
 
             av_reader = av.open(full_path)
+            is_video = True if av_reader.streams.video else False
 
-            if av_reader.streams.video:
-                video_reader = VideoReader(full_path, "video")
+            if is_video:
+                av_frames, vr_frames = [], []
+                av_pts, vr_pts = [], []
+                # get av frames
                 for av_frame in av_reader.decode(av_reader.streams.video[0]):
-                    vr_frame = next(video_reader)
-
-                    assert float(av_frame.pts * av_frame.time_base) == approx(vr_frame["pts"], abs=0.1)
+                    av_frames.append(torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1))
+                    av_pts.append(av_frame.pts * av_frame.time_base)
+                av_reader.close()
 
-                    av_array = torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1)
-                    vr_array = vr_frame["data"]
-                    mean_delta = torch.mean(torch.abs(av_array.float() - vr_array.float()))
+                # get vr frames
+                video_reader = VideoReader(full_path, "video")
+                for vr_frame in video_reader:
```

Comment:
Is there any advantage of have two separate for loops? We could just use one like here: https://github.com/pytorch/vision/blob/e3767f8aa51921460c89e1d396ccce097cebc9d5/test/test_video_gpu_decoder.py#L33 and do something like line 84 on a frame by frame basis and not have to store a list of frames.

Reply:
Theoretically, having separate loops, and closing the file pointers between the readers should increase the stability of the tests as there is no instance in which there are multiple pointers to the same file.\TBH figured it's one of those things that can't hurt, but might increase stability

Reply:
Do you suspect that being the reason for the tests being flaky?

Reply:
Not 100% sure, but I was told it was a good practice, especially when dealing with C++ memcpys.